Sources/Kafka/Configuration/KafkaConsumerConfiguration.swift

```diff
@@ -25,30 +24,20 @@ public struct KafkaConsumerConfiguration {
 
     /// A struct representing different back pressure strategies for consuming messages in ``KafkaConsumer``.
     public struct BackPressureStrategy: Sendable, Hashable {
-        enum _BackPressureStrategy: Sendable, Hashable {
-            case watermark(low: Int, high: Int)
-        }
-
-        let _internal: _BackPressureStrategy
-
-        private init(backPressureStrategy: _BackPressureStrategy) {
-            self._internal = backPressureStrategy
-        }
-
         /// A back pressure strategy based on high and low watermarks.
         ///
         /// The consumer maintains a buffer size between a low watermark and a high watermark
         /// to control the flow of incoming messages.
         ///
         /// - Parameter low: The lower threshold for the buffer size (low watermark).
         /// - Parameter high: The upper threshold for the buffer size (high watermark).
-        public static func watermark(low: Int, high: Int) -> BackPressureStrategy {
-            return .init(backPressureStrategy: .watermark(low: low, high: high))
+        @available(*, deprecated, message: "Use MessageOptions to control backpressure")
```

Comment:
I don't think we need to depreciate stuff here. We aren't 1.0.0 yet so let's just remove it!

Reply:
Removed deprecated code.